### ISSUE TITLE
[A11ySev3] [Programmatic Access -Dashboards with Grafana - New Dashboard-Add Variable]: Logical heading level structure is broken as "Monitor | Dashboards with Grafana | New dashboard" is under <h2> while "query1" is under <h1>.

### ISSUE THREAD

**Issue by LinglingTong** [grafana/grafana#118765]
## Steps to reproduce

Repro Steps: 
  
 
1. Open 'Azure Monitor dashboards with Grafana' in Azure Portal
2. "Monitor | Dashboards with Grafana | Gallery" screen appears with already Dashboards with Grafana tab selected.  
3. Navigate to the all the controls available on Dashboards with Grafana blade like tool bar , controls under All Dashboards tab.  
4. Navigate New menu present in the tool bar and select New Dashboard.  
5. Navigate to "Add visualization" button and hit ENTER key. "Select data source" popup will appear.  
6. Navigate to the TestData list item select it.   
7. Now navigate to the tool bar and select back to dashboard control and select. Now select setting control in the tool bar. 
8. Navigate to Variables tab and select new Variable button. 
9. Navigate to all control under Variable tab and New Variables dialog. 
10. Observe the heading levels assigned to "Monitor | Dashboards with Grafana | New dashboard" and "Query1” or not.

**Actual Result:**
The "Monitor | Dashboards with Grafana | New dashboard" heading is assigned an `<h2>` tag, while "Query1" is assigned an `<h1>` tag. This breaks the logical sequencing of heading levels, as a child or sub-section heading ("Query1") is incorrectly given a higher priority than its parent heading.  
**Observation**:  
The logical structure of headings is not maintained. A parent heading ("Monitor | Dashboards with Grafana | New dashboard") is defined as "`<h2>`", while its child heading ("Query1") is defined as `<h1>`. This inconsistency disrupts the semantic hierarchy of the page and affects screen reader users trying to navigate using heading modes.  
**Issue1**: Same issue reproduces for the "Monitor | Dashboards with Grafana | New dashboard" heading  and "Settings" heading by invoking 'Settings' button.  
**Navigation**: Open https://portal.azure.com/#view/Microsoft_Azure_Monitoring/AzureMonitoringBrowseBlade/~/AzureGrafanaReactView  > Select New button button > Invoke New dashboard control> Select “ Add visualization” button > Select testdata list item> Select Back to dashboard in the tool bar > Select settings button and invoke it > Observe the heading levels assigned to "Monitor | Dashboards with Grafana | New dashboard" and "Settings”.  

**Expected Result:**
All headings should follow a logical and hierarchical structure. For instance, the parent heading "Monitor | Dashboards with Grafana | New dashboard" should be assigned an `<h1>` tag, and its child heading "Query1" should be assigned an `<h2>` tag. This ensures that the relationship between parent and child headings is conveyed programmatically, allowing screen reader users to navigate and understand the content seamlessly.

Since we cannot change the heading of Azure Portal, could GL change the heading to `<h2>` in Grafana?

**Relevant WCAG Criteria:** [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html) (Level A)

## Environment

- **Grafana version:** 12.2.0
- **Data source type & version:**
- **User OS & Browser:** Version 25H2 (OS Build 26200.7093) / Chrome Version 142.0.7444.177 (Official Build) (64-bit)
- **Others:**

**Comment by LinglingTong** [comment]
<img width="1918" height="1090" alt="Image" src="https://github.com/user-attachments/assets/21d4e35b-934e-49d2-a886-3eb13e457bb6" />
<img width="1919" height="1092" alt="Image" src="https://github.com/user-attachments/assets/87bf3070-4c1c-48e4-b3d1-1be4c847d13b" />

**Comment by LinglingTong** [comment]
I have discussed with accessibility team and they agreed to downgrade the issue to Sev3. Updated the issue accordingly.

**Comment by ashharrison90** [comment]
@LinglingTong i don't think this is something that can be easily fixed on the grafana side. grafana will have `h1` elements to denote page headings. when embedding in azure, any of those `h1`s will break the logical structure of the azure page.
